FreeRTOS/FreeRTOS
# [WIN32-MingW Demo] TickType_t width is defined based on compiler type.(32bit/64bit)

[WIN32-MingW Demo] TickType_t width is defined based on compiler type.(32bit/64bit)

Description
-----------

32bit `TickType_t` is used if compiler is MinGW32. 64bit `TickType_t` is used if compiler is MinGW64.

[Reason of change]
Before this change, 32bit `TickType_t` was always used in MinGW demo. It was inefficient for 64bit compiler. In addition, MinGW64 reported warnings for the cast operation between `TickType_t` and `void *` pointer because of different width. 64bit `TickType_t` should be used instead of 32bit if compiler is 64bit.

[Additional change]
Even if 64bit `TickType_t` is used, MinGW64 still reported warnings for the cast operation between 32bit variable and pointer type. To solve that, `uint32_t` has been replaced by `UBaseType_t` in some functions and `%u` has been replaced by `%llu` in `printf()`.

[Related forum topic]
This PR is derived by the following topic in the forum.
[64bit application on FreeRTOS windows port](https://forums.freertos.org/t/64bit-application-on-freertos-windows-port/19110)

[Limitation]
There are still same warnings in TraceRecorder module because 32bit variables are used as address there. They cannot be modified because TraceRecorder is out of FreeRTOS.

Test Steps
-----------

1. Build WIN32-MingW Demo by MinGW64 on **Debug** configuration and on **Debug_CodeCoverage** configuration.
2. Confirm there is no warning for the cast operation. (Except for TraceRecorder)
3. Run executable on both configurations.
4. Confirm they are running successfully.
5. Confirm code coverage is kept after change.

6. Those tests are performed by also MinGW32 as regression tests.

Checklist:
----------



Related Issue
-----------

If this PR is accepted, I will send another PR for similar changes for MSVC-Demo.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
